### PR TITLE
Week 5 assignments

### PR DESCRIPTION
**Pull request by adnan-dedhawala**
### 5.1
**To create 3 callback functions and execute them using 1) async await 2) generator functions**

- Created a function that calls a paginated API and takes page number as a parameter
- Created a async function that in a loop calls the API function 3 times
- Created a async generator function that calls API function 3 times

### 5.2
**Write a function called vowelCount which accepts a string and returns a map where the keys arenumbers and the values are the count of the vowels in the string.**

- Created a function that uses map to store vowels as keys and count of vowels as value to those keys

### 5.3
** Write a function called hasDuplicate which accepts an array and returns true or false if that arraycontains a duplicate**

- Created a function that converts array to set and if the set size equals to original array length , than the function returns false else true

### 5.4
**Create a simple Javascript function code for addition, subtraction, and multiplication of 2numbers and write the corresponding Jest based tests for it**

- Created a object that returns function to add, subract and multiply numbers
- using jest wrote test cases to verify these functions 
- wrote test cases for success , failure and invalid inputs

### CODE REVIEW

File: Week-5/5.1/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    });
+};
+
+const asyncAwaitTask = async () => {
+  let counter = 1;
+  while (counter <= 3) {
+    let characters = await getRickandMortyCharacter(counter);
+    console.log("async await", characters);
+    counter++;
+  }
```
Comment (arqam-pesto) [inline Week-5/5.1/index.js]: Use Promise.all to resolve all at once. Loop resolution will cause issues if anything errors out.
Wrap it in try-catch block

File: Week-5/5.4/Calculator.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const Calculator = {
+  sum: (...args) => {
+    if (args.some(val => isNaN(val))) return "invalid input";
```
Comment (arqam-pesto) [inline Week-5/5.4/Calculator.js]: check for truthy values for validation, `If(val)`
and `!` for negative validation

**Review by arqam-pesto — CHANGES_REQUESTED** [review]

File: Week-5/5.4/Calculator.js (hunk 2 above)
Comment (adnan-dedhawala) [reply Week-5/5.4/Calculator.js]: added check for falsy values

File: Week-5/5.1/index.js (hunk 1 above)
Comment (adnan-dedhawala) [reply Week-5/5.1/index.js]: added try-catch block

**Review by arqam-pesto — APPROVED** [review]

**Pull request merged**